src/library/scala/concurrent/Future.scala

```diff
@@ -608,9 +608,13 @@ object Future {
    * @return          the `Future` of the `TraversableOnce` of results
    */
   def sequence[A, M[X] <: TraversableOnce[X]](in: M[Future[A]])(implicit cbf: CanBuildFrom[M[Future[A]], A, M[A]], executor: ExecutionContext): Future[M[A]] = {
-    in.foldLeft(successful(cbf(in))) {
-      (fr, fa) => for (r <- fr; a <- fa) yield (r += a)
-    } map (_.result())
+    if (in.isEmpty) {
```

Comment:
speaking of which, that should be documented, i.e. what methods are safe to
call twice.

-- 
Cheers,
√